### PR TITLE
Plugin API cleanup refactor

### PR DESCRIPTION
**Pull request by SpacemanPaul**
Move product method out of Plugin API and replaced with helper method `product_for_plugin`.

Also removed some unused test code from pq plugin.

### CODE REVIEW

**Review by Kirill888 — APPROVED** [review]

File: libs/stats/odc/stats/model.py [hunk 1]
```diff
@@ -522,93 +522,94 @@ def rgba(self, xx: xr.Dataset) -> Optional[xr.DataArray]:
...
-            naming_conventions_values=naming_conventions_values,
-            explorer_path=explorer_path,
-            inherit_skip_properties=inherit_skip_properties,
-            preview_image_ows_style=preview_image_ows_style,
-            classifier=classifier,
-            maturity=maturity,
-            collection_number=collection_number,
-            nodata=nodata,
+            version=version_dashed,
+            version_dashed=version_dashed,
```
Comment (tebadi) [inline libs/stats/odc/stats/model.py:582]: Do we need to keep both `version` and `version_dashed `now that they point to the same value?

**Review by tebadi — APPROVED** [review]

File: libs/stats/odc/stats/model.py (hunk 1 above)
Comment (SpacemanPaul) [reply libs/stats/odc/stats/model.py:582]: Probably not.  Just taking trying to keep PRs as small and reviewable as possible.

Comment (SpacemanPaul) [reply libs/stats/odc/stats/model.py:582]: It's a question of whether there are any scripts anywhere that use `{version_dashed}` in their location templates.

Comment (tebadi) [reply libs/stats/odc/stats/model.py:582]: We don't provide `version_dashed ` in our K8s stats calls.  If it's not used in odc-stats then it's probably safe to delete.

**Pull request merged**